crates/dojo-project/src/migration/mod.rs

```diff
@@ -151,49 +151,40 @@ trait Deployable: Declarable {
 #[async_trait]
 impl Declarable for ClassMigration {
     async fn declare(&self, account: &SingleOwnerAccount<SequencerGatewayProvider, LocalWallet>) {
-        let (flattened_class, casm_class_hash) =
-            prepare_contract_declaration_params(&self.artifact_path).unwrap();
-
-        let result = account.declare(Arc::new(flattened_class), casm_class_hash).send().await;
-        match result {
-            Ok(result) => {
-                println!(
-                    "Declared `{}` class at transaction: {:#x}",
-                    self.class.name, result.transaction_hash
-                );
-            }
-            Err(error) => {
-                if error.to_string().contains("already declared") {
-                    println!("{} class already declared", self.class.name)
-                } else {
-                    panic!("Problem declaring {} class: {error}", self.class.name);
-                }
-            }
-        }
+        declare(self.class.name.clone(), &self.artifact_path, account).await;
     }
 }
 
 #[async_trait]
 impl Declarable for ContractMigration {
     async fn declare(&self, account: &SingleOwnerAccount<SequencerGatewayProvider, LocalWallet>) {
-        let (flattened_class, casm_class_hash) =
-            prepare_contract_declaration_params(&self.artifact_path).unwrap();
+        declare(self.contract.name.clone(), &self.artifact_path, account).await;
+    }
+}
 
-        let result = account.declare(Arc::new(flattened_class), casm_class_hash).send().await;
+async fn declare(
+    name: String,
+    artifact_path: &PathBuf,
+    account: &SingleOwnerAccount<SequencerGatewayProvider, LocalWallet>,
+) {
+    let (flattened_class, casm_class_hash) =
+        prepare_contract_declaration_params(artifact_path).unwrap();
 
-        match result {
-            Ok(result) => {
-                println!(
-                    "Declared `{}` contract at transaction: {:#x}",
-                    self.contract.name, result.transaction_hash
-                );
-            }
-            Err(error) => {
-                if error.to_string().contains("already declared") {
-                    println!("{} contract already declared", self.contract.name)
-                } else {
-                    panic!("Problem declaring {} contract: {error}", self.contract.name);
-                }
+    if account.provider().get_class_by_hash(casm_class_hash, BlockId::Pending).await.is_ok() {
+        println!("{} class already declared", name)
```

Comment:
missing `return` here?

```suggestion
        println!("{} class already declared", name);
        return;
```

Reply:
it's implicit but better to be explicit

Reply:
It will be the implicit value for the `if` block, but will not return the function early and still run the rest because you don't put them in the `else` block tho?